File: SeLeAs/LeetCode/Top 100 Liked/bestTimetoBuyandSellStock.py
from typing import List

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy = prices[0]
        max_profit = 0
        for price in prices[1:]:  # start from the second price
            buy = min(buy, price)
            max_profit = max(max_profit, price - buy)
        return max_profit

File: SeLeAs/LeetCode/Top Interview 150/Array_String/BestTimetoBuyandSellStock.java
package Array_String;

class BestTimetoBuyandSellStock {
    public int maxProfit(int[] prices) {
        if (prices == null || prices.length == 0) return 0;

        int buy = prices[0];
        int maxProfit = 0;

        for (int i = 1; i < prices.length; i++) {
            buy = Math.min(buy, prices[i]);
            maxProfit = Math.max(maxProfit, prices[i] - buy);
        }

        return maxProfit;
    }
}